plugins/poll/spec/lib/poll_spec.rb

```diff
@@ -337,4 +338,259 @@
       )
     end
   end
+
+  describe ".serialized_voters" do
+    context "with a regular poll" do
+      let(:post) { post_with_regular_poll }
+      let(:poll) { post.polls.first }
+      let(:poll_options) { poll.poll_options }
+
+      let(:votes) do
+        {
+          user => [poll_options.first.digest],
+          user_2 => [poll_options.second.digest],
+          user_3 => [poll_options.first.digest],
+        }
+      end
+
+      let(:expected_voters) do
+        {
+          poll_options.first.digest => [
+            UserNameSerializer.new(user).serializable_hash,
+            UserNameSerializer.new(user_3).serializable_hash,
+          ],
+          poll_options.second.digest => [UserNameSerializer.new(user_2).serializable_hash],
+        }
+      end
+      let(:expected_paginated_voters) { [expected_voters] }
+
+      before do
+        votes.each_pair { |user, options| DiscoursePoll::Poll.vote(user, post.id, "poll", options) }
+      end
+
+      it "returns all serialized voters" do
+        voters = DiscoursePoll::Poll.serialized_voters(poll)
+        expect(voters).to eq(expected_voters)
+      end
+
+      it "correctly paginates voters" do
+        page = 1
+        loop do
+          expected = expected_paginated_voters[page - 1]
+          opts = { page:, limit: 2 }.with_indifferent_access
+
+          voters = DiscoursePoll::Poll.serialized_voters(poll, opts)
+          expect(voters).to eq(expected)
+
+          break unless voters
+          page += 1
+        end
+      end
+    end
```

Comment:
Apologies, one last amendment please: we'd like to avoid some brain gymnastics in tests if possible. Can you do something like the following?

```
      it "correctly paginates voters" do
        voters = DiscoursePoll::Poll.serialized_voters(poll, { page: 1, limit: 2 })
        expect(voters).to eq(
          {
            poll_options.first.digest => [
              UserNameSerializer.new(user).serializable_hash,
              UserNameSerializer.new(user_3).serializable_hash,
            ],
          },
        )

        voters = DiscoursePoll::Poll.serialized_voters(poll, { page: 2, limit: 2 })
        expect(voters).to eq(
          { poll_options.first.digest => [UserNameSerializer.new(user_2).serializable_hash] },
        )
      end
```

Reply:
I also thought about doing this but wasn't sure what you favored. I'll change it, no problem.

Reply:
(One test has failed, but it seems unrelated...)